src/Compiler/Driver/CompilerImports.fs

```diff
@@ -2255,7 +2255,7 @@ and [<Sealed>] TcImports
             r: AssemblyResolution
         ) : Async<(_ * (unit -> AvailableImportedAssembly list)) option> =
         async {
-            do! Cancellable.UseToken()
+            use! _holder = Cancellable.UseToken()
```

Comment:
this is called only from `RegisterAndImportReferencedAssemblies`, either sequentially or in parallel, depending on settings. Should be sufficient to only do it once, there.